picasso-layoutlib-compat/src/main/java/com/squareup/picasso3/LayoutLibThreadFactory.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.picasso3
+
+import android.annotation.SuppressLint
+import android.os.Handler_Delegate
+import java.util.concurrent.ThreadFactory
+
+/**
+ * LayoutLib requires assistance bridging the gap to work with ThreadFactory.
+ * This workaround ensures the Handler_Delegate executes all threads that are created
+ * by Picasso allowing RequestHandler's the chance to fulfill a request synchronously.
+ */
+@SuppressLint("SyntheticAccessor")
+class LayoutLibThreadFactory : ThreadFactory {
+
+  init {
+    if (layoutLibEnvironment) {
+      // Handler_Delegate sendMessageAtTime drops messages that
```

Comment:
```suggestion
      // Otherwise, Handler_Delegate.sendMessageAtTime drops messages without a callback.
```

Also, explain what thread owns this callback and why it's needed?